Packs/EDL/Integrations/EDL/EDL.py

```diff
@@ -216,14 +217,51 @@ def ips_to_ranges(ips: Iterable, collapse_ips: str):
     Returns:
         Set. a list to Ranges or CIDRs.
     """
+    invalid_ips = []
+    valid_ips = []
 
-    if collapse_ips == COLLAPSE_TO_RANGES:
-        ips_range_groups = IPSet(ips).iter_ipranges()
-        return ip_groups_to_ranges(ips_range_groups)
+    for ip in ips:
+        if is_valid_ip(ip):
+            valid_ips.append(ip)
+        else:
+            invalid_ips.append(ip)
 
+    if collapse_ips == COLLAPSE_TO_RANGES:
+        ips_range_groups = IPSet(valid_ips).iter_ipranges()
+        collapsed_list = ip_groups_to_ranges(ips_range_groups)
     else:
-        cidrs = IPSet(ips).iter_cidrs()
-        return ip_groups_to_cidrs(cidrs)
+        cidrs = IPSet(valid_ips).iter_cidrs()
+        collapsed_list = ip_groups_to_cidrs(cidrs)
+
+    collapsed_list.update(invalid_ips)
+    return collapsed_list
+
+
+def is_valid_ip(ip: str) -> bool:
```

Comment:
let's check the structure and only have maximum, 2 checks per IP.
not sure how scalable it is.
@DeanArbel wdyt?

Reply:
I agree.
I think it's better to drop the `generic` implementation of `ip_groups_to_ranges` and `ips_to_ranges` by passing an extra argument - the type of expected `IP` / `CIDR` - that way the code doesn't need to be made of 4 `try/except`.

Reply:
Fixed